### PR TITLE
Enforce that the ttl is non-zero

### PR DESCRIPTION
**Pull request by Raynos**
In the protocol doc the ttl must be non-zero.

cc @abhinav @prashant I hope your python/go implementation
also enforces this.

r: @jcorbin @rf @shannili

### CODE REVIEW

**Comment by ShanniLi** [comment]
lgtm

File: node/v2/call.js [hunk 1]
```diff
@@ -139,6 +148,12 @@ function writeCallReqInto(body, buffer, offset) {
     // flags:1 -- filled in later after argsrw
     offset += bufrw.UInt8.width;
 
+    if (body.ttl === 0) {
```
Comment (rf) [inline node/v2/call.js]: Is this coerced to a positive integer somewhere? Shouldn't we be checking if it's > 0? What if the user doesn't pass in a ttl at all? Then it's certainly not exactly 0, but it is an invalid ttl.

Comment (Raynos) [inline node/v2/call.js]: True dat.

**Comment by Raynos** [comment]
Adding gaurding against negative ttl.

**Comment by abhinav** [comment]
@junchaowu what's the Python behavior around this?

**Pull request merged**